psst-gui/src/webapi/client.rs

```diff
@@ -26,27 +26,33 @@ use std::{
 };
 use ureq::{Agent, Request, Response};
 
-use super::cache::WebApiCache;
+use super::{cache::WebApiCache, local::LocalTrackManager};
 
 pub struct WebApi {
     session: SessionService,
     agent: Agent,
     cache: WebApiCache,
     token_provider: TokenProvider,
+    local: Option<LocalTrackManager>
 }
 
 impl WebApi {
     pub fn new(
         session: SessionService,
         proxy_url: Option<&str>,
         cache_base: Option<PathBuf>,
+        username: Option<String>
     ) -> Self {
         let agent = default_ureq_agent_builder(proxy_url).unwrap().build();
         Self {
             session,
             agent,
             cache: WebApiCache::new(cache_base),
             token_provider: TokenProvider::new(),
+            local: match username {
+                Some(u) => LocalTrackManager::new(u),
+                _ => None
+            }
```

Comment:
Again `map` I personally find preferable here, but maybe we should re-think how the manager is initialized, see above.